pkg/ui/workspaces/cluster-ui/src/api/databases/getTableMetadataApi.ts

```diff
@@ -0,0 +1,111 @@
+// Copyright 2024 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+import useSWR from "swr";
+
+import { StoreID } from "../../types/clusterTypes";
+import { fetchDataJSON } from "../fetchData";
+import {
+  APIV2ResponseWithPaginationState,
+  SimplePaginationState,
+} from "../types";
+
+const TABLE_METADATA_API_PATH = "api/v2/table_metadata/";
+
+export enum TableSortOption {
+  NAME = "name",
+  REPLICATION_SIZE = "replicationSize",
+  RANGES = "ranges",
+  LIVE_DATA = "liveData",
+  COLUMNS = "columns",
+  INDEXES = "indexes",
+  LAST_UPDATED = "lastUpdated",
+}
+
+export type TableMetadata = {
+  db_id: number;
+  db_name: string;
+  table_id: number;
+  schema_name: string;
+  table_name: string;
+  replication_size_bytes: number;
+  range_count: number;
+  column_count: number;
+  index_count: number;
+  percent_live_data: number;
+  total_live_data_bytes: number;
+  total_data_bytes: number;
+  store_ids: number[] | null;
```

Comment:
is this nullable / Are you seeing cases where this is null instead of an empty array?